### ISSUE TITLE
Usage of PLSuperExpCutoff4FGL in a fit

### ISSUE THREAD

**Issue by lmohrmann** [gammapy/gammapy#2373]
I'm wondering how `PLSuperExpCutoff4FGL` is supposed to be used in a fit. The problem I see is that the parameter `expfactor` comes with a unit that depends on the value of `index_2`, and I don't think there's a way to handle this properly in a fit (or is there?).

Or is the solution to use `PLSuperExpCutoff3FGL` when running a fit?

@cdeil @kaoriinakashima @yuwunwong I think you implemented this, right?

**Comment by adonath** [comment]
I guess you are referring to the following problem:
```python3
from gammapy.spectrum.models import PLSuperExpCutoff4FGL
from astropy import units as u

ecpl = PLSuperExpCutoff4FGL()
ecpl.index_2.value = 2.2
ecpl(1 * u.TeV)
```
Which fails with a unit error.

The `PLSuperExpCutoff4FGL` was mainly added for catalog access and plotting. In principle you can transform the `expfactor` parameter of the `4FGL` model into the `E_C` parameters of the `3FGL`, but the problem is, that you cannot propagate the error correctly, because the covariance info is missing. I think that's why we decided to introduce the new `4FGL` model.

I'm not sure how to handle this. Making the unit dependent on the value of another parameters seems technically not easy to solve. This leaves the question, whether we should introduce `non-fittable` models or an `alwaysfrozen` attribute to model parameters, which does not allow to change the value of a model parameter. 

On the other hand, you can just use the `PLSuperExpCutoff3FGL`  model for fitting, which is the same model with a slightly different parametrisation.

**Comment by lmohrmann** [comment]
I'd say it's OK to have this model mainly for plotting. There should then be a remark in the docstring for this model saying that `PLSuperExpCutoff3FGL` should be used for fitting (maybe even with a short explanation of how to do the parameter transformation). And maybe, as you suggest, introduce a `non_fittable` flag such that the fit can immediately return a useful error message?

**Comment by cdeil** [comment]
@QRemy applied a fix in #2395.

@lmohrmann - OK now? Or can it be improved further somehow?

**Comment by cdeil** [comment]
@lmohrmann - Please re-open this issue or file a new issue or PR if something should be improved.
